optimism/src/keccak/witness.rs

```diff
@@ -15,6 +15,28 @@ use super::{
     DIM, HASH_BYTELENGTH, QUARTERS, WORDS_IN_HASH,
 };
 
+pub(crate) fn pad_blocks<Fp: Field>(pad_bytelength: usize) -> Vec<Fp> {
+    // Blocks to store padding. The first one uses at most 12 bytes, and the rest use at most 31 bytes.
+    let mut blocks = vec![Fp::zero(); 5];
+    let mut pad = [Fp::zero(); RATE_IN_BYTES];
+    pad[RATE_IN_BYTES - pad_bytelength] = Fp::one();
+    pad[RATE_IN_BYTES - 1] += Fp::from(0x80u8);
```

Comment:
Nit: just an assignment would be a bit faster, as it won't have to compute the addition. Here it does one addition + one new allocation for the result of the addition.

Reply:
This is because `0x81` is a possibility (adding 0x80 to 0x01). Are you suggesting that a set of if-else would be slightly faster?